### PR TITLE
[RPS-350] Made Doc type facets consistent

### PR DESCRIPTION
**Pull request by jacsamell**
Made them use the configurable doc type labels.
Also fixed a bug where taxonomy terms would repeat if the
taxonomy key was invalid.

### CODE REVIEW

File: site/src/main/java/uk/nhs/digital/common/components/FacetComponent.java [hunk 1]
```diff
@@ -3,20 +3,45 @@
...
+
+        private TaxonomyWrapper(Taxonomy taxonomy) {
+            this.taxonomy = taxonomy;
+        }
+
+        public String getValueName(String key) {
+            Category taxonomyCategory = taxonomy.getCategoryByKey(key);
+            if (taxonomyCategory == null) {
+                log.error("No taxonomy for key: " + key);
+                return "Unknown Option: " + key;
```
Comment (wojtek-oledzki) [inline site/src/main/java/uk/nhs/digital/common/components/FacetComponent.java]: Something to check with PO. 
In terms of presentation, maybe it's better to return nothing, or do some guessing like replace "-" with " " etc...

File: repository-data/application/src/main/resources/hcm-content/content/documents/administration/document-types.yaml [hunk 2]
```diff
@@ -70,9 +70,9 @@
     - publicationsystem:dataset
     - publicationsystem:publication
     resourcebundle:messages:
-    - Series
-    - Data Set
-    - Statistical Publication
+    - labels.series
+    - labels.dataset
+    - labels.publication
```
Comment (wojtek-oledzki) [inline repository-data/application/src/main/resources/hcm-content/content/documents/administration/document-types.yaml]: can you add "Archive" as well pls

Comment (wojtek-oledzki) [reply repository-data/application/src/main/resources/hcm-content/content/documents/administration/document-types.yaml]: oh wait, it's not merged yet :D, never mind.

File: site/src/main/java/uk/nhs/digital/common/components/FacetComponent.java (hunk 1 above)
Comment (jacsamell) [reply site/src/main/java/uk/nhs/digital/common/components/FacetComponent.java]: Have updated after input from Karen, happy for it not be be pretty as it should only happen with bad data on test/uat and shouldn't ever happen on prod.

**Pull request merged**